### PR TITLE
feat: TOOLS-3092 Add schema diff and list-schemas commands

### PR DESCRIPTION
**Pull request by a-spiker**
Implements `schema-diff` to compare Aerospike configuration schemas between two versions and `list-schemas` to list available schema versions. Includes comprehensive property extraction for arrays and oneOf structures, improved error handling, and extensive unit tests for command workflows and edge cases.

### CODE REVIEW

File: cmd/schema_diff.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+	if props, ok := schema["properties"].(map[string]interface{}); ok {
+		extractPropertiesRecursive(props, "", properties)
+	}
+
+	return properties
+}
+
+// PropertyInfo holds information about a schema property
+type PropertyInfo struct {
```
Comment (sunilvirus) [inline cmd/schema_diff.go:153]: I am wondering why bother extracting into a temporary structure. 
why can we store the whole sub-element and compare directly json structures.

**Review by sunilvirus — CHANGES_REQUESTED** [review]
Suggested some changes. we can discuss offline.

File: cmd/list_schemas.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			logger.Debug("Running list-schemas command")
+
+			// Load schema map
+			schemaMap, err := schema.NewSchemaMap()
+			if err != nil {
+				return fmt.Errorf("failed to load schema map: %w", err)
+			}
+
+			// Get all version keys (exclude README)
```
[the remark this answers is not in the document]
Comment (a-spiker) [reply cmd/list_schemas.go:34]: Thats an old functionality, i am not sure why that folder has readme..   filtered the readme from schema map

File: cmd/schema_diff.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+	}
+}
+
+// compareSchemasDetailed compares two sets of schema properties and returns detailed differences
+func compareSchemasDetailed(props1, props2 map[string]PropertyInfo, showDetails, showDeprecated bool, filterPath string) []string {
+	var diffs []string
+
+	// Get all unique keys
+	allKeys := make(map[string]struct{})
+	for k := range props1 {
```
[the remark this answers is not in the document]
Comment (a-spiker) [reply cmd/schema_diff.go:248]: currently there is no deprecated field or anything to mark deprecation. This is purely experimental, I was thinking of having. I'll remove the logic around this.
